pkgs/applications/networking/instant-messengers/dino/default.nix

```diff
@@ -62,6 +62,10 @@ stdenv.mkDerivation rec {
     librsvg
   ];
 
+  patches = [
+    ./signal-2.3.3.patch
```

Comment:
Does the following apply?

```
(fetchpatch {
  url = "https://github.com/dino/dino/commit/fbd70ceaac5ebbddfa21a580c61165bf5b861303.patch";
  sha256 = "<INSERT_REAL_SHA256_HERE>";
})
```

Reply:
```nix
patches = [
  (fetchpatch {
    url = "https://github.com/dino/dino/commit/fbd70ceaac5ebbddfa21a580c61165bf5b861303.patch";
    sha256 = "1l13pwal1b11hiky0p1x27acyqzhzii63b99vf89i54zjwir9wm8";
  })
];
```
does not apply:
```
patching file plugins/signal-protocol/CMakeLists.txt
File plugins/signal-protocol/libsignal-protocol-c is not a regular file -- refusing to patch
1 out of 1 hunk ignored -- saving rejects to file plugins/signal-protocol/libsignal-protocol-c.rej
builder for '/nix/store/k45xhlgbksgc5rripx2k4vwagnmjcc4x-dino-0.1.0.drv' failed with exit code 1
error: build of '/nix/store/k45xhlgbksgc5rripx2k4vwagnmjcc4x-dino-0.1.0.drv' failed
```
Is there a way to ignore the part of the patch that changes the submodule commit or to fix this another way?

Reply:
You can add `includes = [ "plugins/signal-protocol/CMakeLists.txt" ];` or `excludes = [ "plugins/signal-protocol/libsignal-protocol-c" ];` (note that the hash will change).

Reply:
Thanks for the tip!